grid/crm/lists/maps/sendgrid.suma

```diff
@@ -0,0 +1,52 @@
+profile = "crm/lists@1.0"
+provider = "sendgrid"
+
+map GetLists {
+    http GET "/v3/marketing/lists" {
+        security "bearer_token"
+
+        request {
+            query {
+                page_token = input.page,
+            }
+        }
+
+        response 200 "application/json" {
+            map result {
+                lists = body.result.map((list) => {
+                    return {
+                        listId: list.id,
+                        name: list.name,
+                        length: list.contact_count
+                    }
+                })
+                nextPage = body._metadata.next ? body._metadata.next.match('page_token=([a-z0-9-]+)')[1] : null
+            }
+        }
+    } 
+}
+
+// If the contact already exists in the system, any entries submitted via this endpoint
```

Comment:
suggestion: I do not think that it is right that the map implements`upsert` but the name of the use case is `AddContactsToList`. I suggest to change use case name to `AddOrUpdateContacts`.

I checked that MailChimp provider which supports `upsert` as well.

Reply:
The use case is "add a contact to a contact list", and the map implements that; for this provider, it's using an endpoint which has upsert semantics, but there's actually no updating going on.

Reply:
OK, i understand. There is still one disproportion that should be harmonised across providers.  AddContactsToList can either:
- do nothing if already existing email is being added (which is this case)
- fail if already existing email is being added

We should have test for it.